### ISSUE TITLE
Request

### ISSUE THREAD

**Issue by fennectech** [HamletDuFromage/aio-switch-updater#25]
Could you remove the version number so the filename is always the same making updates drag and drop when using forwarders

**Comment by HamletDuFromage** [comment]
Can't really do that because of the way I auto-update the app. It can't overwrite itself so a different name is needed. There is a way to make it work, but it'd involve a bit of rewrite.

I'll keep it in mind though

**Comment by fennectech** [comment]
Could you make a NRO forwarder that runs it no mater what the filename is?

**Comment by HamletDuFromage** [comment]
you know what, I'll look into it

**Comment by vgmoose** [comment]
I was able to get around NRO-self-updating issues in hb-appstore by stopping using the switch romfs and moving to a in-memory alternative for loading assets (in this case, [resinfs](https://gitlab.com/4TU/resinfs), which came from the wiiu hb scene)

I had also tried unmounting the romfs prior to updating the NRO but this gave mixed results. Using a loader sounds like a pretty decent solution too, in our use-case i didn't want to maintain that however.

**Comment by HamletDuFromage** [comment]
Self updating isn't a problem, an easy solution would simply be to pack a NRO in romfs and launch it then overwrite the app NRO with it. The problem would be to update to that new file from within the current NRO. Basically I did it in a dumb way the first time around because I didn't know any better, and now I'm kinda stuck

**Comment by fennectech** [comment]
Chain load to another NRO that does the fine NRO replacement?

**Comment by fennectech** [comment]
Basicly.  Download and place an update.nro which is run. Now that the origional NRO is free replace it and return to the now updated version and delete the update.nro

**Comment by HamletDuFromage** [comment]
>Chain load to another NRO that does the fine NRO replacement?
yes. problem is that the new version will be called aio-switch-updater.nro and the current app is expecting a aio-switch-updater-vx.x.x.nro so the update experience won't be smooth

**Comment by fennectech** [comment]
Have it remove the other filename too.

**Comment by vgmoose** [comment]
The old version can't remove the other one, the user would have to be stuck with two versions until the next launch of the new one to clean it up (I think, if understanding right)

**Comment by HamletDuFromage** [comment]
yeah you're right, it's exactly that. So now the question is : is it worth it?

**Comment by fennectech** [comment]
The update.nro can clean up the old version.    Its an ugly migration but it would work.

**Comment by fennectech** [comment]
The idea is  step 1
1. AIO updater downloads and places update.nro onto the sdcard
2. AIO updater runs update.nro
3. Updat.nro deletes aio updater old version
4. Update.nro deletes the version with the version number
5. Update.nro places the new aio updater on the sdcard 
6. update.nro returns to aio updater
7. Aio updater deletes update.nro

**Comment by HamletDuFromage** [comment]
Done in the latest update. You can find a forwarder below
[All-in-OneSwitchUpdater_NSP_Forwarder[058ED764CE7A0000].nsp.zip](https://github.com/HamletDuFromage/aio-switch-updater/files/5889756/All-in-OneSwitchUpdater_NSP_Forwarder.058ED764CE7A0000.nsp.zip)
